### PR TITLE
[PM-8137] Set initial FIDO 2 user verification state

### PR DESCRIPTION
**Pull request by SaintPatrck**
## 🎟️ Tracking

https://bitwarden.atlassian.net/browse/PM-8137

## 📔 Objective

When a FIDO 2 intent is received the FIDO user verification state is set to unverified so that user verification settings will be evaluated and respected if necessary.

When the user unlocks their vault after the first intent is received the FIDO user verification state is set to verified.

## ⏰ Reminders before review

- Contributor guidelines followed
- All formatters and local linters executed and passed
- Written new unit and / or integration tests where applicable
- Used internationalization (i18n) for all UI strings
- CI builds passed
- Communicated to DevOps any deployment requirements
- Updated any necessary documentation or informed the documentation team

## 🦮 Reviewer guidelines

- 👍 (`:+1:`) or similar for great changes
- 📝 (`:memo:`) or ℹ️ (`:information_source:`) for notes or general info
- ❓ (`:question:`) for questions
- 🤔 (`:thinking:`) or 💭 (`:thought_balloon:`) for more open inquiry that's not quite a confirmed
  issue and could potentially benefit from discussion
- 🎨 (`:art:`) for suggestions / improvements
- ❌ (`:x:`) or ⚠️ (`:warning:`) for more significant problems or concerns needing attention
- 🌱 (`:seedling:`) or ♻️ (`:recycle:`) for future improvements or indications of technical debt
- ⛏ (`:pick:`) for minor or nitpick changes

### CODE REVIEW

File: app/src/main/java/com/x8bit/bitwarden/MainViewModel.kt [hunk 1]
```diff
@@ -218,6 +220,7 @@ class MainViewModel @Inject constructor(
             }
 
             fido2CredentialRequestData != null -> {
+                fido2CredentialManager.isUserVerified = false
```
Comment (shannon-livefront) [inline app/src/main/java/com/x8bit/bitwarden/MainViewModel.kt:223]: Should we include some documentation about why this is always set to false?

File: app/src/main/java/com/x8bit/bitwarden/data/autofill/fido2/manager/Fido2CredentialManager.kt [hunk 2]
```diff
@@ -11,6 +11,12 @@ import com.x8bit.bitwarden.data.autofill.fido2.model.Fido2ValidateOriginResult
  */
 interface Fido2CredentialManager {
 
+    /**
+     * Returns true when the user has performed an explicit verification action. E.g., biometric
+     * verification, device credential verification, or vault unlock.
+     */
+    var isUserVerified: Boolean
```
Comment (shannon-livefront) [inline app/src/main/java/com/x8bit/bitwarden/data/autofill/fido2/manager/Fido2CredentialManager.kt:18]: Dumb question, but for my own knowledge, this is only for the case where the user doesn't have screen lock enabled on their device so they haven't been verified by the time they get to the app, right?

File: app/src/test/java/com/x8bit/bitwarden/ui/auth/feature/vaultunlock/VaultUnlockViewModelTest.kt [hunk 3]
```diff
@@ -871,6 +876,38 @@ class VaultUnlockViewModelTest : BaseViewModelTest() {
...
+                isBiometricLogin = true,
+            ),
+        )
+
+        verify { fido2CredentialManager.isUserVerified = true }
+    }
+
+    @Suppress("MaxLineLength")
+    @Test
+    fun `on ReceiveVaultUnlockResult should set FIDO 2 user verification state to verified when result is not Success`() {
```
Comment (shannon-livefront) [inline app/src/test/java/com/x8bit/bitwarden/ui/auth/feature/vaultunlock/VaultUnlockViewModelTest.kt:897]:
```suggestion
    fun `on ReceiveVaultUnlockResult should set FIDO 2 user verification state to not verified when result is not Success`() {
```

File: app/src/main/java/com/x8bit/bitwarden/data/autofill/fido2/manager/Fido2CredentialManager.kt (hunk 2 above)
Comment (SaintPatrck) [reply app/src/main/java/com/x8bit/bitwarden/data/autofill/fido2/manager/Fido2CredentialManager.kt:18]: Not a dumb question at all. It's not exclusive to device verification, but is only relevant in the context of FIDO operations. It represents a broader verification state that also includes BW methods of authentication like unlocking the vault or entering their MP to view a protected cipher. The expectation is that when we add BW authentication methods to this flow we will update `isUserVerified` based on the result.

Comment (shannon-livefront) [reply app/src/main/java/com/x8bit/bitwarden/data/autofill/fido2/manager/Fido2CredentialManager.kt:18]: Awesome thanks!

File: app/src/main/java/com/x8bit/bitwarden/MainViewModel.kt [hunk 4]
```diff
@@ -220,6 +220,9 @@ class MainViewModel @Inject constructor(
             }
 
             fido2CredentialRequestData != null -> {
+                // Set the user's verification status when a new FIDO 2 request is received to force
+                // explicit verification if the user's vault is unlocked when the request is
```
[the remark this answers is not in the document]
Comment (SaintPatrck) [reply app/src/main/java/com/x8bit/bitwarden/MainViewModel.kt:224]: No. The scenario we're handling here is if the user wants to register/authenticate a passkey and their vault is already unlocked but the relying party requires/prefers explicit user verification. If the user's vault is **locked** when this intent is received they have to unlock it, which marks them verified when done successfully.

[the remark this answers is not in the document]
Comment (shannon-livefront) [reply app/src/main/java/com/x8bit/bitwarden/MainViewModel.kt:224]: Ahhh, got it, thanks!

**Review by shannon-livefront — APPROVED** [review]
Looks good!

**Pull request merged**